### PR TITLE
chore(cleanup): idk what to do so here

### PR DESCRIPTION
**Pull request by archbear**
## Summary by CodeRabbit

- **Refactor**
  - Renamed `DBPruner` to `pruner` and adjusted method signatures.
  - Updated logic within the `Start` method for improved index pruning.

- **New Features**
  - Implemented a more generalized approach to function parameters, enhancing flexibility and maintainability.

- **Bug Fixes**
  - Modified condition checks and event handling for improved accuracy and performance.

These changes aim to streamline functionality and improve system robustness.

### CODE REVIEW

**Review by ocnc2 — APPROVED** [review]
service[_, _, _, _, _, _, _, lgtmT, _, _, _, _, _, _,]

File: mod/state-transition/pkg/core/state_processor_slashing.go [hunk 1]
```diff
@@ -29,10 +29,7 @@ import (
 //
 //nolint:lll
 func (sp *StateProcessor[
-	BeaconBlockT, BeaconBlockBodyT, BeaconBlockHeaderT,
-	BeaconStateT, BlobSidecarsT, ContextT,
-	DepositT, Eth1DataT, ExecutionPayloadT, ExecutionPayloadHeaderT,
-	ForkT, ForkDataT, ValidatorT, WithdrawalT, WithdrawalCredentialsT,
+	_, _, _, BeaconStateT, _, _, _, _, _, _, _, _, _, _, _,
```
[the remark this answers is not in the document]
Comment (ocnc2) [reply mod/state-transition/pkg/core/state_processor_slashing.go:32]: thanks for the suggestion @coderabbitai

[the remark this answers is not in the document]
Comment (ocnc2) [reply mod/state-transition/pkg/core/state_processor_slashing.go:32]: u up u up u up

[the remark this answers is not in the document]
Comment (ocnc2) [reply mod/state-transition/pkg/core/state_processor_slashing.go:32]: 🥕

[the remark this answers is not in the document]
Comment (archbear) [reply mod/state-transition/pkg/core/state_processor_slashing.go:32]: @ocnc2 and @coderabbitai sitting in a tree, k-i-s-s-i-n-g

[the remark this answers is not in the document]
Comment (ocnc2) [reply mod/state-transition/pkg/core/state_processor_slashing.go:32]: LOL wtf

[the remark this answers is not in the document]
Comment (ocnc2) [reply mod/state-transition/pkg/core/state_processor_slashing.go:32]: its learning

**Review by itsdevbear — APPROVED** [review]

**Pull request merged**